### PR TITLE
Implement rotation in RelativeMode, move deltaing to the end of a pipeline

### PR DESCRIPTION
**Pull request by AbstractQbit**
Rotation is done via a transform matrix, along with a sensitivity scaling.
Moving delta creation to the end allows filters to work with position instead of velocity, which they were not designed to handle.

### CODE REVIEW

**Comment by X9VoiD** [comment]
Am not against the change, but just wanna say that this breaks [PrecisionControl](https://github.com/X9VoiD/VoiDPlugins/blob/master/PrecisionControl/PrecisionControl.cs#L50).

Will clean up the code tho since we don't have to check if it's RelativeOutputMode anymore. Will also fix all other filters since only PrecisionControl handles relative data lol

**Comment by AbstractQbit** [comment]
The idea was that you won't have to fix all the other filters, since now they should work the same as in absolute mode

File: OpenTabletDriver/Settings.cs [hunk 1]
```diff
@@ -192,6 +192,13 @@ public float YSensitivity
             get => _yS;
         }
 
+        [JsonProperty("RelRotation")]
+        public float RelRotation
+        {
+            set => this.RaiseAndSetIfChanged(ref _relRot, value);
+            get => _relRot;
+        }
```
Comment (InfinityGhost) [inline OpenTabletDriver/Settings.cs:200]: Rename to `RelativeRotation`, avoid shortened names for major properties like this.

**Review by InfinityGhost — CHANGES_REQUESTED** [review]
Looks good, just one rename and I'm happy with it

**Review by InfinityGhost — APPROVED** [review]

File: OpenTabletDriver.Plugin/Output/RelativeOutputMode.cs [hunk 2]
```diff
@@ -32,24 +32,42 @@ public IEnumerable<IFilter> Filters
...
+                UpdateTransformMatrix();
+            }
+            get { return _rotation; }
+        }
+
+        private void UpdateTransformMatrix()
+        {
+            _lastReceived = default;  // Prevents cursor from jumping on sensitivity change
+            _transformationMatrix = Matrix3x2.CreateRotation(
+                (float)(_rotation * System.Math.PI / 180));
```
Comment (InfinityGhost) [inline OpenTabletDriver.Plugin/Output/RelativeOutputMode.cs:61]: Invert rotation to match AbsoluteOutputMode

**Review by InfinityGhost — CHANGES_REQUESTED** [review]

**Review by InfinityGhost — APPROVED** [review]

**Pull request merged**